client/src/test/java/io/pravega/client/segment/impl/SegmentInputStreamTest.java

```diff
@@ -401,4 +417,43 @@ public void testReadWithNegativeEndOffset() {
                 -2, SegmentInputStreamImpl.DEFAULT_BUFFER_SIZE);
     }
     
+    
+    @Test
+    public void testOrderer() throws EndOfSegmentException, SegmentTruncatedException {
+        byte[] data = new byte[]{0, 1, 2, 3, 4, 5, 6, 7, 8, 9};
+        ByteBuffer wireData = createEventFromData(data);
+        
+        TestAsyncSegmentInputStream fakeNetwork1 = new TestAsyncSegmentInputStream(segment, 1);
+        TestAsyncSegmentInputStream fakeNetwork2 = new TestAsyncSegmentInputStream(segment, 2);
+        TestAsyncSegmentInputStream fakeNetwork3 = new TestAsyncSegmentInputStream(segment, 3);
+        TestAsyncSegmentInputStream fakeNetwork4 = new TestAsyncSegmentInputStream(segment, 4);
+        
+        @Cleanup
+        SegmentInputStreamImpl stream1 = new SegmentInputStreamImpl(fakeNetwork1, 0);
+        @Cleanup
+        SegmentInputStreamImpl stream2 = new SegmentInputStreamImpl(fakeNetwork2, 0);
+        @Cleanup
+        SegmentInputStreamImpl stream3 = new SegmentInputStreamImpl(fakeNetwork3, 0);
+        @Cleanup
+        SegmentInputStreamImpl stream4 = new SegmentInputStreamImpl(fakeNetwork4, 0);
+
+        fakeNetwork2.complete(0, new WireCommands.SegmentRead(segment.getScopedName(), 0, false, false, wireData.slice()));
+        fakeNetwork3.complete(0, new WireCommands.SegmentRead(segment.getScopedName(), 0, false, true, wireData.slice()));
+        fakeNetwork4.complete(0, new WireCommands.SegmentRead(segment.getScopedName(), 0, false, true, ByteBuffer.allocate(0)));
+        
+        Orderer o = new Orderer();
+        List<SegmentInputStreamImpl> segments = ImmutableList.of(stream1, stream2, stream3, stream4);
```

Comment:
This test case is deterministic, which has a positive side. I wonder if it would make sense to have a second test case tjhat randomizes the segment or segments that are actually sealed and verify that the order is as expected.

Reply:
I've thought about it, and I'm not sure it makes sense. The orderer does not have any special logic to pick the sealed segments first. It merely has two sets of segments, those it is willing to return and those it is not.
So I think the more generalized test is to ensure that it will return those that are sealed. This test verifies that orderer in fact does return them as expected. But that determination is really made by 'canReadWithoutBlocking' which is really the more important function to test. That is now being covered by the other test that I modified.